Add rate limiting using Reddit RateLimit API response headers

jfl88 opened the pull request:
I took @braboobssiere's development and also applied it to the handle function as well.

With this change all functions (search, edit, delete) will look at the RateLimit feedback from Reddit API and use it to set the timeout function period. When the response from Reddit is 5 queries remaining or less, it will wait for the reset period to expire before executing again.

jfl88 committed:
Change 5sec delay to 7sec

jfl88 committed:
Add rate limit recognition

jfl88 committed:
correct bracket issue

jfl88 committed:
fix bracket issue

jfl88 committed:
fixing brackets

jfl88 committed:
final bracket fix

jfl88 committed:
increase default timeout

jfl88 committed:
add debug

jfl88 committed:
add rate-limited to handle function

jfl88 committed:
Update powerdeletesuite.js

jfl88 committed:
Update powerdeletesuite.js

replace handle .then with .complete

jfl88 committed:
Update powerdeletesuite.js

jfl88 committed:
Update powerdeletesuite.js

jfl88 committed:
Update powerdeletesuite.js

jfl88 committed:
Update powerdeletesuite.js

jfl88 committed:
Merge branch 'j0be:master' into master

jfl88 commented:
Note: this pull request doesn't update version information or anything, I will leave that up to @j0be to decide how to handle any versioning or repo maintenance.

jokarl commented:
Nice addition! Cleaning up an account with a lot of comments takes a long time when it requires manual intervention in an alert dialog every time it does not respect the rate limit.

fidelisrafael commented:
This is awesome! I was about to try to implement something in this fashion, since I want to let this running while I am not around, and having to confirm on the confirm popup was getting on my way. 

I changed the one-line script to point to your branch and changes:

```
javascript:(function() { window.bookmarkver = '1.4'; var isReddit = document.location.hostname.split('.').slice(-2).join('.') === 'reddit.com'; var isOverview = !! document.location.href.match(/\/overview\b/i); if (isReddit && isOverview) { var cachBustUrl = 'https://raw.githubusercontent.com/j0be/PowerDeleteSuite/194be0e6d795052d6d1b9b35138233f170da29e8/powerdeletesuite.js?' + (new Date().getDate()); fetch(cachBustUrl).then(function(response) { return response.text(); }).then(function(data) { var script = document.createElement('script'); script.id = 'pd-script'; script.innerHTML = data; document.getElementsByTagName('head')[0].appendChild(script); }).catch(function() { alert('Error retrieving PowerDeleteSuite from GitHub'); }); } else if (confirm('This script can only be run from your own user profile on Reddit. Would you like to go there now?')) { document.location = 'https://old.reddit.com/user/me/overview'; } else { alert('Please go to your Reddit profile before running this script'); } })();
```

Thanks @jfl88

fidelisrafael commented on powerdeletesuite.js:
```diff
@@ -714,21 +729,27 @@ var pd = {
-                pd.actions.page.next();
-              }
+              const timeout = pd.helpers.getRateLimitTimeout(xhr);
+              console.log('timeout: ' + timeout);
```
I want to suggest this to be changed to: `console.debug` so one can hide these log lines from the main console's output...

jfl88 committed:
Update cache bust URL for PowerDeleteSuite

jfl88 committed:
Enhance timeout logging with current date

Updated console log statements to include the current date when logging timeout values.

jfl88 committed:
Update bookmarklet.js source URL in user script
